train.py

```diff
@@ -1,216 +1,228 @@
-
-def normalize(data):
-    data_max = max(data)
-    data_min = min(data)
-    data_norm = max(data_max,abs(data_min))
-    return data / data_norm
-
-def main(args):
-    '''Ths is a similar Tensorflow/Keras implementation of the LSTM model from the paper:
-        "Real-Time Guitar Amplifier Emulation with Deep Learning"
-        https://www.mdpi.com/2076-3417/10/3/766/htm
-
-        Uses a stack of two 1-D Convolutional layers, followed by LSTM, followed by 
-        a Dense (fully connected) layer. Three preset training modes are available, 
-        with further customization by editing the code. A Sequential tf.keras model 
-        is implemented here.
-
-        Note: RAM may be a limiting factor for the parameter "input_size". The wav data
-          is preprocessed and stored in RAM, which improves training speed but quickly runs out
-          if using a large number for "input_size".  Reduce this if you are experiencing
-          RAM issues. Also, you can use the "--split_data" option to divide the data by the
-          specified amount and train the model on each set. Doing this will allow for a higher
-          input_size setting (more accurate results).
-        
-        --training_mode=0   Speed training (default)
-        --training_mode=1   Accuracy training
-        --training_mode=2   Extended training (set max_epochs as desired, for example 50+)
-    '''
-
-    name = args.name
-    if not os.path.exists('models/'+name):
-      os.makedirs('models/'+name)
-    else:
-      print("A model folder with the same name already exists. Adding timestamp for uniqueness.")
-      timestamp = datetime.datetime.now().strftime("%Y%m%d%H%M%S")
-      os.makedirs(f'models/{name}_{timestamp}')
-
-    train_mode = args.training_mode     # 0 = speed training, 
-                                        # 1 = accuracy training 
-                                        # 2 = extended training
-    batch_size = args.batch_size 
-    test_size = 0.2
-    epochs = args.max_epochs
-    input_size = args.input_size
-
-    # TRAINING MODE
-    if train_mode == 0:         # Speed Training
-        learning_rate = 0.01 
-        conv1d_strides = 12    
-        conv1d_filters = 16
-        hidden_units = 36
-    elif train_mode == 1:       # Accuracy Training (~10x longer than Speed Training)
-        learning_rate = 0.01 
-        conv1d_strides = 4
-        conv1d_filters = 36
-        hidden_units= 64
-    else:                       # Extended Training (~60x longer than Accuracy Training)
-        learning_rate = 0.0005 
+    Modes:
+        Speed training (default)
+        Accuracy training
+        Extended training (set max_epochs as desired, for example 50+)
+    """
+    print(f"Training started")
+
+    timestamp = datetime.datetime.now().strftime("%Y-%b-%d-%H%M%S").lower()
+    name, in_files, out_files = IO.validate_source(source_dir)
+    out_dir = f'{name}_{timestamp}'
+    os.makedirs(f'models/{out_dir}')
```

Comment:
Prev version would fail on name collision w folder, which made it difficult to re-run similar models back to back. This removes collision risk via timestamps in the folder names. Ugly but useful.